command/catalog/list/services/catalog_list_services.go

```diff
@@ -77,19 +84,48 @@ func (c *cmd) Run(args []string) int {
 			return 1
 		}
 		if catalogNode != nil {
-			services = make(map[string][]string, len(catalogNode.Services))
+			services = make(map[string]*serviceInfo, len(catalogNode.Services))
 			for _, s := range catalogNode.Services {
-				services[s.Service] = append(services[s.Service], s.Tags...)
+				services[s.Service] = &serviceInfo{
+					address: s.Address,
+					port:    s.Port,
+					tags:    s.Tags,
+				}
+			}
+		}
+	} else if c.service != "" {
+		catalogServices, _, err := client.Catalog().Service(c.service, "", nil)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error print service: %s", err))
+			return 1
+		}
+
+		if catalogServices != nil {
+			services = make(map[string]*serviceInfo, len(catalogServices))
+			for _, s := range catalogServices {
+				services[s.ServiceName] = &serviceInfo{
```

Comment:
This will only show info for one instance of the target service (because there's only room for one instance in the services map) which doesn't really make sense for the whole service, as each instance will have a separate address/port combination. This should probably be a list of addr/ports for each instance.

Reply:
You're right. Thanks.

Reply:
You're right. Thanks.